### PR TITLE
Fix bz67675

### PR DESCRIPTION
**Pull request by markt-asf**
Fixes https://bz.apache.org/bugzilla/show_bug.cgi?id=67675

providing as a PR to allow review before commit

### CODE REVIEW

**Comment by michael-o** [comment]
There is also AES128 with OpenSSL anf should be tried as well.

File: java/org/apache/tomcat/util/net/jsse/PEMFile.java [hunk 1]
```diff
@@ -64,8 +66,40 @@ public class PEMFile {
...
+        // 1.2.840.113549.2.11
+        OID_TO_PRF.put("2a864886f70d020b", "HmacSHA512");
+        // 1.2.840.113549.2.12
+        OID_TO_PRF.put("2a864886f70d020c", "HmacSHA512/224");
+        // 1.2.840.113549.2.13
+        OID_TO_PRF.put("2a864886f70d020d", "HmacSHA512/256");
+    }
 
-    private static final String PBES2 = "PBES2";
+    private static final Map<String,Algorithm> OID_TO_ALGOITHM = new HashMap<>();
```
Comment (michael-o) [inline java/org/apache/tomcat/util/net/jsse/PEMFile.java:94]: Typo

**Comment by michael-o** [comment]
The more ASN.1 I see in Tomcat the more I think is that we should import Kerby ASN.1 at some point in time...

File: test/org/apache/tomcat/util/net/jsse/TestPEMFile.java [hunk 2]
```diff
@@ -118,4 +127,17 @@ private String getPath(String file) throws IOException {
...
     }
+
+
+    @Test
+    public void testListSecretKeyFactories() {
+        for (Provider provider : Security.getProviders()) {
+            System.out.println( provider );
+            for (Provider.Service service : provider.getServices()) {
+                if ("Cipher".equals( service.getType())) {
+                    System.out.println( service.getAlgorithm() );
```
Comment (michael-o) [inline test/org/apache/tomcat/util/net/jsse/TestPEMFile.java:138]: Are the spaces on purpose?

**Comment by michael-o** [comment]
Thanks for the hard work. I will try this tomorrow/Wednesday!

**Comment by michael-o** [comment]
Built from your branch and I can confirm that this works with a selfsigned and one issued by our enterprise CA.

**Comment by markt-asf** [comment]
Added a test key using AES128 (test passed without additional code changes).
Fixed typo.
Removed debug code that wasn't intended to be part of the commit.
I might be persuaded to include Kerby ASN.1 although I'm not a fan of additional dependencies. We'd need to shade it if we did. Something to consider alongside the whether we want to change how we handle the Commons packages.

**Comment by michael-o** [comment]
Yes, through shading only to make them private. Looking at the vast amount of code you have written already I bet that ther are also others which would like to use such a PEM to Java Keystore converter. I consider it a pain everytime to have the tripe of key, cert and password and them being forced just for Java to convert it.

**Pull request merged**